internal/npm_install/node_module_library.bzl

```diff
@@ -32,8 +32,8 @@ def _node_module_library_impl(ctx):
     # declarations are a subset of sources that are decleration files
     declarations = depset([f for f in ctx.files.srcs if f.path.endswith(".d.ts")])
 
-    # transitive_declarations are all direct & transitive decleration files
-    transitive_declarations = depset()
+    # transitive_declarations are all .d.ts files in srcs plus those in direct & transitive dependencies
+    transitive_declarations = depset(transitive = [declarations])
     for src in ctx.attr.srcs:
```

Comment:
does it really make sense that node_module_library might have non-src entries in srcs? Seems to me we should only walk deps[] to collect the transitive

Reply:
per slack, removed the walk over srcs

Reply:
yup. srcs should only be files or filegroup so they have no "typescript" provider to walk over